includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -1171,13 +1171,21 @@ public function maybe_filter_gateway_title( $title, $id ) {
 	 * @return void
 	 */
 	public function set_payment_method_title_for_email( $order ) {
+		$payment_gateway = wc_get_payment_gateway_by_order( $order );
+
+		if ( ! empty( $payment_gateway ) && self::GATEWAY_ID !== $payment_gateway->id || ! WC_Payments_Features::is_upe_legacy_enabled() ) {
```

Comment:
@mdmoore I was passing by as I'm currently reviewing [tests for this enhancement](https://github.com/Automattic/woocommerce-payments/pull/7331). With filtering out any UPE type other than legacy UPE, we only allow legacy UPE gateways to update the email's payment method title. Effectively, deferred intent creation UPE now inserts "Credit card / debit card" in the email, vs "Visa debit card" in legacy UPE. Thus, for deferred UPE users starting `6.5.1`, the email title differ from what they saw previously, and this group will be increased starting `6.6.0`, when all the legacy UPE stores will be switched to the deferred UPE.

Reply:
Thanks for bringing this up. [This function was introduced](https://github.com/Automattic/woocommerce-payments/pull/6351) to ensure that `Popular Payment Methods` did not appear in emails as the payment method. My understanding was that the `Popular Payment Methods` payment method title is only set for Legacy UPE. This was my reason for limiting this change to Legacy UPE with hopes that we could likely remove this function down the line. Are you aware of other UPEs that will set the gateway title to `Popular Payment Methods`?

 >deferred intent creation UPE now inserts "Credit card / debit card" in the email, vs "Visa debit card" in legacy UPE.

I think prior to the above mentioned PR(#6351), `Credit card / debit card` always appeared. As a side effect of that PR, the card network and card type were used instead. This explains why now only Legacy UPE is displaying the network and card type as other UPE types are now back to the previous behavior. I'm also not sure of the desired behavior. Once we find out we can ensure consistency.

Reply:
Your argumentation makes total sense and I totally understand your approach, @mdmoore! The current difference between UPEs is fine for me and I'll proceed to review tests with the assumption that we keep everything as is and, as it turns out, remove this logic once we migrate everyone to deferred intent UPE, but in case we feel it needs further alignment - feel free to follow this up and I'm always there for you to review if any further changes will be needed in the future. 👍